Added A2.4 & A2.5 from The Nurbs Book

Nodli opened the pull request:
Hi, just added two functions from The Nurbs Book.
Both functions have been tested on Python2.7.15 and Python3.7.0.
Should tests be included in the project ?
Let me know if there are any issues

orbingol commented:
@Nodli thank you very much for adding algorithms A2.4 and A2.5. It would be really good to add tests and additionally, compare them with respect to existing function outputs, if possible.

orbingol commented:
@Nodli just forgot to comment on the tests. All tests are included inside `tests/` folder. You may check the documentation for more details: http://nurbs-python.readthedocs.io/en/latest/install.html#testing

Nodli commented:
Thanks for the answer.
I added the test_helpers.py file because there was none inside the tests/ folder.
A2.4 and A2.5 implementations are tested by comparison with A2.2 and A2.3.
I could not find the tests for the other functions inside helpers.py so I wrote some based on values from The Nurbs Book.

Just for information there is an error in A2.5 in The Nurbs Book as it gives wrong values when the span of the basis function to evaluate is not the span of the knot. I corrected this error in my implementation.

orbingol commented:
@Nodli thank you very much for all the coding and bug fixing. I would like to comment on one thing about the tests: you don't need to explicitly call the test functions at the end of `test_helpers.py` file. `pytest` will automatically discover all the tests under `tests` directory, files and functions starting with `test_`. The only requirement is running the `pytest` executable at the root (i.e. not inside the `tests` directory). It feels more simple and more convenient in this way.

The other thing I would like to make sure before merging is the new functions producing similar values with the existing ones (up to some tolerance, e.g. defined by `GEOMDL_DELTA` constant in the test file).  This is something I do for the `CurveEvaluator` and `CurveEvaluator2` classes. They contain different algorithms for evaluating n-th order curve derivatives.  You may take a look at [test_bsplien_curve_deriv1](https://github.com/orbingol/NURBS-Python/blob/master/tests/test_BSpline_Curve3D.py#L164) as a reference.

For the basis functions, I believe that the output of  `helpers.basis_function` and `helpers.basis_function_one` +  the derivative ones should be somewhat correlated and they could be tested towards each other. I have never used A2.4 and A2.5 directly, please correct me if I am wrong.

Nodli commented:
Thanks for the comment about pytest, I never used it so it's nice to learn about it. Function calls have been removed from the file.

I have changed the tests to include the tolerance value, I agree that it's better now.

Yes the results are the same. A2.4 and A2.5 are not very useful expect for tests and I don't remember them being used anywhere in the first half of the book actually. I just included the tolerance value on thoses tests, but I think I already tested them against each other, let me know if you are expecting something else on that.
